### PR TITLE
layers: Add option flags to spirv-opt pass

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Add the same option flags `spirv-val` was using to also be used for `spirv-opt` otherwise it will fail in `spirv-opt` (see original issue #3442)

Add positive tests to prevent regression

~~@muusbolla this tests passes for me (locally, curious if the wider CI machines will catch the issue)~~

~~In effort to reproduce the issue you are facing on the issue, trying to figure out what else I might be missing... do you have the raw SPIR-V that is fed to `vkCreateShaderModule` (truncated or not is fine) in order to not rely on glslang for reproducing the issue~~

### CODE REVIEW

File: tests/vkpositivelayertests.cpp [hunk 1]
```diff
@@ -16188,3 +16188,140 @@ TEST_F(VkPositiveLayerTest, ShaderPointSizeStructMemeberWritten) {
...
+    TEST_DESCRIPTION("Reproduces issue 3442 where spirv-opt fails to set layout flags options using Vulkan 1.0");
+    // https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/3442
+
+    m_errorMonitor->ExpectSuccess();
+    if (InstanceExtensionSupported(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME)) {
+        m_instance_extension_names.push_back(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+    } else {
+        printf("%s Did not find VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME; skipped.\n", kSkipPrefix);
+        return;
+    }
```
Comment (ncesario-lunarg) [inline tests/vkpositivelayertests.cpp:16202]: I know this was probably copied from another test, but would it be possible to change this to something like
```cpp
if (!AddRequiredExtensions(VK_KHR_UNIFORM_BUFFER_STANDARD_LAYOUT_EXTENSION_NAME) ||
    !AddRequiredExtensions(VK_EXT_SCALAR_BLOCK_LAYOUT_EXTENSION_NAME)) ...
```
followed by
```cpp
if (!AreRequestedExtensionsAvailable()) ...
```
after `InitFramework`? I will try to refactor the rest of the tests to use this sooner than later.

**Comment by sfricke-samsung** [comment]
(going to keep conversation here on the PR instead of the issue)

@muusbolla so the `VkShaderObj` is just a fancy wrapper around `VkShaderModule` that will call `vkCreateShaderModule` and then destroy it when it goes out of scope

So I just copied and pasted the GLSL shader you had in the issue (which didn't show which values were being read/written to)

I tried changing the test to read from both the Spec Constant and the UBO and also changed the default value via a `VkSpecializationInfo` as well and **I finally see the issue!**

`spirv-opt` has the same flags `spirv-val` such as `--scalar-block-layout` but we were not setting them when calling `spirv-opt` which is what you are seeing.... going to quick clean up what I have and push a fix along with these tests to prevent regress

Thanks again for raising this issue, I am REALLY surprised this was actually like this, probably forever, and glad someone shinned the light on it

File: layers/shader_validation.cpp [hunk 2]
```diff
@@ -2269,6 +2269,10 @@ bool CoreChecks::ValidatePipelineShaderStage(const PIPELINE_STATE *pipeline, con
             }
         }
 
+        // both spirv-opt and spirv-val will use the same flags
+        spvtools::ValidatorOptions options;
+        AdjustValidatorOptions(device_extensions, enabled_features, options);
```
Comment (ncesario-lunarg) [inline layers/shader_validation.cpp:2274]: Wow, kind of amazing this hasn't caused issues sooner!

**Review by ncesario-lunarg — APPROVED** [review]
Thanks @sfricke-samsung and @muusbolla!

**Pull request merged**